csm/conf/cleanup.py

```diff
@@ -59,3 +63,51 @@ def files_directory_cleanup(self):
     def web_env_file_cleanup(self):
        Log.info(f"Replacing {const.CSM_WEB_DIST_ENV_FILE_PATH}_tmpl {const.CSM_WEB_DIST_ENV_FILE_PATH}")
        Setup._run_cmd(f"cp -f {const.CSM_WEB_DIST_ENV_FILE_PATH}_tmpl {const.CSM_WEB_DIST_ENV_FILE_PATH}")
+
+    def cleanup_ldap_config(self):
+        #Delete cortxuser schema
+        filelist = glob.glob('/etc/openldap/slapd.d/cn=config/cn=schema/*cortxuser.ldif')
+        for schemafile in filelist:
+            try:
+                os.remove(schemafile)
+            except Exception as e:
+                Log.error(f"Failed to delete cortxuser schema: {e}")
+                raise CsmSetupError("Failed to delete cortxuser schema.")
+        #Delete permission attributes
+        try:
+            self._search_delete_permission_attr("olcDatabase={2}mdb,cn=config", "olcAccess")
+        except Exception as e:
+            Log.error(f"Failed to delete the ldap permission attributes: {e}")
+            raise CsmSetupError("Failed to delete ldap permission aatributes.")
+        #Restart slapd service
+        Setup._run_cmd('systemctl restart slapd')
+
+    def _search_delete_permission_attr(self, dn, attr_to_delete):
+        conn = ldap.initialize("ldapi://")
+        conn.sasl_non_interactive_bind_s('EXTERNAL')
+        ldap_result_id = conn.search_s(dn, ldap.SCOPE_BASE, None, [attr_to_delete])
+        total = 0
+        # Below will count the entries
+        for result1,result2 in ldap_result_id:
```

Comment:
Variable names have to be more user friendly

Reply:
done